# Joint Position Controller
$(arg namespace):

   joint1_position_controller:
      type: position_controllers/JointPositionController
      joint: joint1

   joint2_position_controller:
      type: position_controllers/JointPositionController
      joint: joint2

   joint3_position_controller:
      type: position_controllers/JointPositionController
      joint: joint3

   joint4_position_controller:
      type: position_controllers/JointPositionController
      joint: joint4

   joint5_position_controller:
      type: position_controllers/JointPositionController
      joint: joint5

   joint6_position_controller:
      type: position_controllers/JointPositionController
      joint: joint6

   joint7_position_controller:
      type: position_controllers/JointPositionController
      joint: joint7

   # Joint Group Position Controller
   pos_joint_group_controller:
      type: position_controllers/JointGroupPositionController
      joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - joint6
      - joint7

   # Joint Trajectory Controllers
   # For detailed explanation of parameters see http://wiki.ros.org/joint_trajectory_controller

   pos_joint_traj_arm_controller:
      type: position_controllers/JointTrajectoryController
      joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - joint6
      state_publish_rate:  125
      action_monitor_rate: 125

   pos_joint_traj_controller:
      type: position_controllers/JointTrajectoryController
      joints:
      - joint1
      - joint2
      - joint3
      - joint4
      - joint5
      - joint6
      - joint7
      state_publish_rate:  125
      action_monitor_rate: 125

   # Publish all joint states

   joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 125

   # Gripper controller
   open_close_gripper_controller:
      type: open_close_gripper_controller/GripperOpenCloseController
      joint: gripper_left_joint
      min_position: 0.021